src/raster/rasterband.rs

```diff
@@ -17,37 +17,49 @@ use ndarray::Array2;
 
 use crate::errors::*;
 
-/// Resampling algorithms, map GDAL defines
+/// Resampling algorithms used throughout various GDAL raster I/O operations.
+///
+/// # Example
+///
+/// ```rust
+/// use gdal::Dataset;
+/// # fn main() -> gdal::errors::Result<()> {
+/// use gdal::raster::ResampleAlg;
+/// let ds = Dataset::open("fixtures/tinymarble.tif")?;
+/// let band1 = ds.rasterband(1)?;
+/// let stats = band1.get_statistics(true, false)?.unwrap();
+/// // Down-sample a image using cubic-spline interpolation
+/// let buf = band1.read_as::<f64>((0, 0), ds.raster_size(), (2, 2), Some(ResampleAlg::CubicSpline))?;
+/// // In this particular image, resulting data should be close to the overall average.
+/// assert!(buf.data.iter().all(|c| (c - stats.mean).abs() < stats.std_dev / 2.0));
+/// # Ok(())
+/// # }
+/// ```
 #[derive(Debug, Copy, Clone)]
+#[repr(u32)]
 pub enum ResampleAlg {
     /// Nearest neighbour
-    NearestNeighbour,
+    NearestNeighbour = GDALRIOResampleAlg::GRIORA_NearestNeighbour,
     /// Bilinear (2x2 kernel)
-    Bilinear,
+    Bilinear = GDALRIOResampleAlg::GRIORA_Bilinear,
     /// Cubic Convolution Approximation (4x4 kernel)
-    Cubic,
+    Cubic = GDALRIOResampleAlg::GRIORA_Cubic,
     /// Cubic B-Spline Approximation (4x4 kernel)
-    CubicSpline,
+    CubicSpline = GDALRIOResampleAlg::GRIORA_CubicSpline,
     /// Lanczos windowed sinc interpolation (6x6 kernel)
-    Lanczos,
+    Lanczos = GDALRIOResampleAlg::GRIORA_Lanczos,
     /// Average
-    Average,
+    Average = GDALRIOResampleAlg::GRIORA_Average,
     /// Mode (selects the value which appears most often of all the sampled points)
-    Mode,
+    Mode = GDALRIOResampleAlg::GRIORA_Mode,
     /// Gauss blurring
-    Gauss,
+    Gauss = GDALRIOResampleAlg::GRIORA_Gauss,
 }
 
-fn map_resample_alg(alg: &ResampleAlg) -> u32 {
```

Comment:
Note: this was not private before, so there was no easy way to map values when calling into lower level `gdal-sys` functions.

Reply:
Also, the return type should have been `GDALRIOResampleAlg::Type`.